### PR TITLE
doc changes for hotfix to NSB.Azure

### PR DESCRIPTION
**Pull request by dbelcham**

### CODE REVIEW

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ - nservicebus/sagas
+ - nservicebus/azure-storage-persistence
+---
+
+
+## Summary
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #74](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/74) using NServiceBus.Azure hotfix release 6.2.5.
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #26](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/26) using the NServiceBus.Azure hotfix release 6.2.4.
```
Comment (SeanFeldman) [inline nservicebus/upgrades/asp-saga-pruning.md]: Should this be a separate block or a bullet list for versions would suffice?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+### Upgrading endpoints
+
+All endpoints using NServiceBus.Azure will need to be upgraded to version 6.2.5 or higher.
+
+
+### Patching data
+
+Saga data stored in Azure will need to be patched using the `IndexPruner` utility which can be downloaded from []()
```
Comment (SeanFeldman) [inline nservicebus/upgrades/asp-saga-pruning.md]: Link is missing as we don't have it yet I presume.
(https://github.com/Particular/IssueDetection/releases/tag/???)

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## Patch steps
+
+ 1. Download the index pruning tool from []() and put it on a computer that has internet access as well as the .NET Framework 4.5.2 installed.
+ 1. Add an Azure Storage connection string to the `IndexPruner.exe.config` file. For example:
```
Comment (SeanFeldman) [inline nservicebus/upgrades/asp-saga-pruning.md]: > Add an Azure Storage connection string

"Add Azure Storage connection string named `sagas`"

We look for a specific connection string.

File: nservicebus/upgrades/asp-6to1.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ related:
 - nservicebus/upgrades/asp-saga-deduplication
 ---
 
-WARNING: Upgrades from NServiceBus.Azure v6.2.3 or lower will need to apply the [saga de-duplication patch](/nservicebus/upgrades/asp-saga-deduplication.md) before completing the remainder of these upgrade steps.
+WARNING: Upgrades from NServiceBus.Azure v6.2.3 or lower will need to apply the [saga de-duplication patch](/nservicebus/upgrades/asp-saga-deduplication.md) and the [saga index patch](/nservicebus/upgrades/asp-saga-pruning.md) before completing the remainder of these upgrade steps.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-6to1.md]: does order matter?

File: nservicebus/upgrades/asp-saga-pruning.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: the title says "6.2.4 to 6.2.5" so why is moving to 6.2.4 mentioned?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+---
+
+
+## Summary
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #74](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/74) using NServiceBus.Azure hotfix release 6.2.5.
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #26](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/26) using the NServiceBus.Azure hotfix release 6.2.4. 
+
+WARNING: When upgrading to NServiceBus.Persistence.AzureStorage Version 1 and above, the following upgrade will need to be performed prior to beginning any other upgrade steps.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: this article is targeting "Azure Storage Persistence upgrade Version 6.2.4 to 6.2.5"  why mention NServiceBus.Persistence.AzureStorage Version   ?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## Patch steps
+
+ 1. Download the index pruning tool from []() and put it on a computer that has internet access as well as the .NET Framework 4.5.2 installed.
+ 1. Add an Azure Storage connection string to the `IndexPruner.exe.config` file. For example:
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: why a config file? if this is a once off thing then a command line parameter is sifficient

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+		</connectionStrings>
+	</configuration>
+	```
+ 1. Copy all endpoint dlls to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be prunded in Azure Storage.
+ 1. Open a commandline and run the following command: `IndexPruner.exe`. The tool will output details of the actions that it is taking.
+ 1. Update NServiceBus.Azure dependency to version 6.2.4 or higher in all endpoints that use it and release the updated endpoints.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: if they update to 6.2.4 without doing the above will they get a warning?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #74](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/74) using NServiceBus.Azure hotfix release 6.2.5.
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #26](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/26) using the NServiceBus.Azure hotfix release 6.2.4. 
+
+WARNING: When upgrading to NServiceBus.Persistence.AzureStorage Version 1 and above, the following upgrade will need to be performed prior to beginning any other upgrade steps.
+
+
+### How to know if a system may be affected
+
+This bug will affect any system that uses sagas and NServiceBus.Azure v6.2.4 or lower.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: perhaps 

> This bug will affect any system that has ever used sagas in NServiceBus.Azure v6.2.4 or lower.

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+ 1. Add an Azure Storage connection string to the `IndexPruner.exe.config` file. For example:
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
+		</connectionStrings>
+	</configuration>
+	```
+ 1. Copy all endpoint dlls to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be prunded in Azure Storage.
+ 1. Open a commandline and run the following command: `IndexPruner.exe`. The tool will output details of the actions that it is taking.
```
Comment (SimonCropp) [inline nservicebus/upgrades/asp-saga-pruning.md]: do endpoints need to be stopped during this execution?

File: nservicebus/upgrades/asp-saga-pruning.md (hunk 2 above)
Comment (dbelcham) [inline nservicebus/upgrades/asp-saga-pruning.md]: Yep. Once we create the release package of the tool I'll update this doc and then move to merge this PR

File: nservicebus/upgrades/asp-6to1.md (hunk 4 above)
Comment (dbelcham) [inline nservicebus/upgrades/asp-6to1.md]: It doesn't but I've updated the text to suggest the order to keep in line with our "do each upgrade in sequence" policy

File: nservicebus/upgrades/asp-saga-pruning.md (hunk 5 above)
Comment (dbelcham) [inline nservicebus/upgrades/asp-saga-pruning.md]: This exists in the 6.2.3 to 6.2.4 doc as well. The v6 to v1 upgrade points people to both of these docs and these warnings are intended to reinforce that people coming from the v6 to v1 doc can't skip these steps during the v6 to v1 upgrade

File: nservicebus/upgrades/asp-saga-pruning.md (hunk 7 above)
Comment (dbelcham) [inline nservicebus/upgrades/asp-saga-pruning.md]: There will be no warning. If they took this route orphaned saga indexes would remain the the Azure Table and the only discernible issue would be increased storage size and, thus, increased storage costs.

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+ - nservicebus/sagas
+ - nservicebus/azure-storage-persistence
+---
+
+
+## Summary
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #74](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/74) using NServiceBus.Azure hotfix release 6.2.5.
+
+WARNING: When upgrading to NServiceBus.Persistence.AzureStorage Version 1 and above, the following upgrade will need to be performed prior to beginning any other upgrade steps.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: @dbelcham - Should we mention the saga de-deduplication patch here in this warning? 

Something like:
WARNING: When upgrading to NServiceBus.Persistence.AzureStorage Version 1 and above, ensure that you have applied the [saga de-duplication patch](/nservicebus/upgrades/asp-saga-deduplication.md) followed by this patch. 

Thoughts?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+
+### Patching data
+
+Saga data stored in Azure will need to be patched using the `IndexPruner` utility which can be downloaded from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284)
+
+
+## Patch steps
+
+ 1. Download the index pruning tool from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284) and put it on a computer that has internet access as well as the .NET Framework 4.5.2 installed.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: How about:

Download the [Index pruning tool](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284) to a computer that has  .NET Framework 4.5.2 installed and also has access to the internet.

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+ 1. Optionally, add an Azure Storage connection string to the `IndexPruner.exe.config` file uses `name="sagas"`. If the connection string is not added to `IndexPruner.exe.config` it will have to be provided as a commandline parameter in step 4 below. Config file example:
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: >Optionally, add an Azure Storage connection string to the `IndexPruner.exe.config` file uses `name="sagas"`.

Can you clarify this sentence? It doesn't read well. 

How about:
The tool needs an Azure Storage connection string. The connection string can be added to the config file as shown below or provided directly as a command line parameter to the tool.

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
+		</connectionStrings>
+	</configuration>
+	```
+ 1. Copy all endpoint dlls to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.
+ 1. Open a commandline and run the following command: `IndexPruner.exe`. If the Azure connection string is not added to the `IndexPruner.exe.config` file in step 2, the command needed to run the Pruner will be `IndexPruner.exe &ltconnectionstringvalue&gt`. While running, the tool will output details of the actions that it is taking.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: Do you need to run this tool with administrator access?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+ 1. Download the index pruning tool from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284) and put it on a computer that has internet access as well as the .NET Framework 4.5.2 installed.
+ 1. Optionally, add an Azure Storage connection string to the `IndexPruner.exe.config` file uses `name="sagas"`. If the connection string is not added to `IndexPruner.exe.config` it will have to be provided as a commandline parameter in step 4 below. Config file example:
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
+		</connectionStrings>
+	</configuration>
+	```
+ 1. Copy all endpoint dlls to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: > Copy all endpoint dlls

is the all the endpoints that need to be patched that's affected by the bug?

Does this imply multiple endpoints or the endpoint that needs to be patched + all its dependencies that are in the bin folder for that endpoint?

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+
+### Upgrading endpoints
+
+All endpoints using NServiceBus.Azure will need to be upgraded to version 6.2.5 or higher.
+
+
+### Patching data
+
+Saga data stored in Azure will need to be patched using the `IndexPruner` utility which can be downloaded from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284)
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: Is there any short introduction on what the IndexPruner actually does?

File: nservicebus/upgrades/asp-saga-pruning.md (hunk 13 above)
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: > While running, the tool will output details of the actions that it is taking.

How about:
The tool will display relevant details about all the actions that are being performed to the command window during the execution.

Comment (dbelcham) [inline nservicebus/upgrades/asp-saga-pruning.md]: No need to run as administrator

File: nservicebus/upgrades/asp-saga-pruning.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+ 1. Download the index pruning tool from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284) to a computer that has the .NET Framework 4.5.2 installed and also has internet access .
+ 1. The `IndexPruner` utility requires an Azure Storage connection string. The connection string can be added to the `IndexPruner.exe.config` file with the `name="sagas"` value or provided as a commandline parameters (shown in step 4).
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
+		</connectionStrings>
+	</configuration>
+	```
+ 1. For each endpoint that is being patched, copy all the assemblies that contain saga type type definitions to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/asp-saga-pruning.md]: How about:
Copy the endpoint dll along with the assemblies that contain saga type type definitions to the same folder as the index pruning tool. If multiple endpoints require this patch, it is ok to add the assemblies for all the endpoints to the same location. This saves time from having to run the tool for each endpoint. These assemblies will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.

**Pull request merged**